src/jetson/face_detector.py

```diff
@@ -0,0 +1,143 @@
+import torch
+
+from src.jetson.models.Retinaface.data.config import cfg_mnet as cfg
+from src.jetson.models.Retinaface.data.config import cfg_inference as infer_params
+from src.jetson.models.utils.transform import BaseTransform
+from src.jetson.models.Retinaface.layers.functions.prior_box import PriorBox
+from src.jetson.models.utils.box_utils import decode, do_nms, postprocess
+
+class FaceDetector:
+    def __init__(self, detector: str, detector_type: str, detection_threshold=0.7, cuda=True, set_default_dev=False):
+        """
+        Creates a FaceDetector object
+        Args:
+            detector: A string path to a trained pth file for a ssd model trained in face detection
+            detector_type: A DetectorType describing which face detector is being used
+            detection_threshold: The minimum threshold for a detection to be considered valid
+            cuda: Whether or not to enable CUDA
+            set_default_dev: Whether or not to set the default device for PyTorch
+        """
+
+        if cuda and torch.cuda.is_available():
+            self.device = torch.device("cuda:0")
+            if set_default_dev:
+                torch.set_default_tensor_type('torch.cuda.FloatTensor')
+        else:
+            self.device = torch.device("cpu")
+            if set_default_dev:
+                torch.set_default_tensor_type('torch.FloatTensor')
+
+        if detector_type == 'ssd':
+            from src.jetson.models.SSD.ssd import build_ssd
+
+            self.net = build_ssd('test', 300, 2)
+            self.model_name = 'ssd'
+            self.net.load_state_dict(torch.load(detector, map_location=self.device))
+            self.transformer = BaseTransform(self.net.size, (104, 117, 123))
+
+        elif detector_type == 'blazeface':
+            from src.jetson.models.BlazeFace.blazeface import BlazeFace
+
+            self.net = BlazeFace(self.device == torch.device("cuda:0"))
+            self.net.load_weights(detector)
+            self.net.load_anchors("models/BlazeFace/anchors.npy")
+            self.model_name = 'blazeface'
+            self.net.min_score_thresh = 0.75
+            self.net.min_suppression_threshold = 0.3
+            self.transformer = BaseTransform(128, None)
+
+        elif detector_type == 'retinaface':
+            from src.jetson.models.Retinaface.retinaface import RetinaFace, load_model
+
+            self.net = RetinaFace(cfg=cfg, phase='test')
+            self.net = load_model(self.net, detector, load_to_cpu=self.device == torch.device("cpu"))
+            self.model_name = 'retinaface'
+            self.image_shape = infer_params["image_shape"]  # (H, W)
+            self.resize = infer_params["resize"]
+            self.transformer = BaseTransform((self.image_shape[1], self.image_shape[0]), (104, 117, 123))
+            priorbox = PriorBox(cfg, image_size=self.image_shape)
+            priors = priorbox.forward()
+            self.prior_data = priors.data.to(self.device)
```

Comment:
Since we're refactoring. we should coonsider creating a base class for `FaceDetector` where each implementation is defined. this will clean up this code and make adding / removing new detector types in the future

Reply:
I agree but I think we can make that a separate task on Trello and we can have a team member work on that